server/src/MessageHandler.ts

```diff
@@ -131,6 +135,46 @@ function getBlockAtPosition(line: number, lines: Array<string>): Block | void {
   return
 }
 
+function getModelOrEnumBlock(
+  blockName: string,
+  lines: string[],
+): Block | void {
+  // get start position of model type
+  const results: number[] = lines
+    .map((line, index) => {
+      if (
+        (line.includes('model') && line.includes(blockName)) ||
+        (line.includes('enum') && line.includes(blockName))
+      ) {
+        return index
+      }
+    })
+    .filter((index) => index !== undefined) as number[]
```

Comment:
It looks like you could put the logic in `map` directly into `filter`

Reply:
How do I get the index when using filter though? I want to get the position of a model or enum with a given name.